scripts/pod_lib_lint.rb

```diff
@@ -162,4 +172,10 @@ def trace(*args)
   STDERR.puts(args.join(' '))
 end
 
+def remove_social_url_from_podspec(spec)
+  podspec_content = File.read(spec)
+  updated_podspec_content = podspec_content.gsub("s.social_media_url = ", "# s.social_media_url = ")
+  File.open(spec, "w") { |file| file.puts updated_podspec_content }
```

Comment:
This modifies the file in place which dirties the workspace. Wouldn't writing to a temporary file be better?

Also:
  * please use parens in function invocations for consistency with the rest of the file
  * prefer `file.write` over `file.puts`. The latter adds an extra newline
  * line length on the line above (shoot for 80)

Reply:
Looks like using a temporary file is not really feasible unless we copy entire workspace:
- if I use a file in a temp dir lint fails with errors like:
```
bundle exec pod lib lint --sources=https://cdn.cocoapods.org/ --include-podspecs={FirebaseCore.podspec,GoogleUtilities.podspec,FirebaseCoreDiagnosticsInterop.podspec,FirebaseCoreDiagnostics.podspec,GoogleDataTransportCCTSupport.podspec,GoogleDataTransport.podspec,FirebaseInstanceID.podspec} /var/folders/mf/06z96qvx2rsgdrhhthyj1sn800lvg7/T/FirebaseInstallations.podspec --platforms=ios

 -> FirebaseInstallations (0.1.0)
    - ERROR | [iOS] file patterns: The `source_files` pattern did not match any file.
    - WARN  | [iOS] file patterns: The `public_header_files` pattern did not match any file.
    - WARN  | [iOS] license: Unable to find a license file
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description
    - ERROR | xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | xcodebuild:  xcodebuild: error: The workspace named "App" does not contain a scheme named "". The "-list" option can be used to find the names of the schemes in the workspace.
```
- even if I use a temp file in the same dir (with intention to remove it afterwards) I get error like
```
bundle exec pod lib lint --sources=https://cdn.cocoapods.org/ --include-podspecs={FirebaseCore.podspec,GoogleUtilities.podspec,FirebaseCoreDiagnosticsInterop.podspec,FirebaseCoreDiagnostics.podspec,GoogleDataTransportCCTSupport.podspec,GoogleDataTransport.podspec,FirebaseInstanceID.podspec} FirebaseInstallations_temp.podspec --platforms=ios

 -> FirebaseInstallations (0.1.0)
    - ERROR | name: The name of the spec should match the name of the file.
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Planning build
    - NOTE  | xcodebuild:  note: Constructing build description
    - ERROR | xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | xcodebuild:               AppHost-FirebaseInstallations-Unit-Tests.app encountered an error (Failed to install or launch the test runner. (Underlying error: The operation couldn’t be completed. (Mach error -308 - (ipc/mig) server died)))
```

So, I wonder, it is a big concern to have a dirty workspace during lint? I can add running `git reset --hard ` on the script completion to avoid interference between different runs of `pod_lib_lint.rb`  within a single job.

Reply:
In this case, I think it's fine to dirty the workspace. Every subjob starts with a fresh clone and nothing about our tests care about the social_media specification anyway.

Reply:
@wilhuff I updated the code style (hope I didn't miss anything).

Reply:
Style changes LGTM

I've sent #3421 which avoids dirtying the workspace while avoiding `git reset`, which is problematic because it can discard local edits to the file.